### DISCUSSION TITLE
Auto redirect user to the app after login if there is the only one app exists

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#2301] [General]
Hey all,

Is there a way to redirect user after login to the app without visiting portal if there is the only one app exists?
I was sure it works that way but now it is not,  I remember - It was working that way before.
In my case visiting the portal after user login is unwanted/unnecessary step which is also requires user action - click on app while I have the only one. Another reason - I don't want my users be able to change their first/last names. 

It would be great to have auto redirect to the app if the only one exists, as it worked before.

Thank you

**Comment by shogunpurple** [comment]
Hi @sergey-dme this functionality was removed, because it was causing a lot of issues and we didn’t want to make it the default. We could possibly make this a configurable property, so it can be turned on or off.

**Reply by sergey-dme** [reply]
Hi @shogunpurple Thanks for the answer. It would be nice to have such option/configurable property 🙏

**Reply by sergey-dme** [reply]
@shogunpurple I found a temporary solution to redirect user to my app after login until that configurable property will be available in bb - using `returnUrl` in login page address 
Example: ` ../builder/auth/login?returnUrl=%252Fapp_d0331c1d4c0346da9624ab30605fb34e`
But currently it is hardly linked to this app id, I mean this is absolute link.. 
Is there a binding for **id** of current app like {{ app_id }} or something exist?
I would use it on my log out button for "Navigate to" action. 

Thank you

**Reply by KernOil** [reply]
@sergey-dme that is exactly what I do. I forward a subdomain to the app's URL that contains its ID. that is the best workaround I have found to bypass the app portal. I'm not really that big of a fan of the app portal I definitely think there should be an option to do a direct login into each application.

**Reply by shogunpurple** [reply]
@AndrewJRamay when you say you aren't a big fan of the app portal, what specifically about it do you think could be improved?

**Reply by sergey-dme** [reply]
@AndrewJRamay I've tried to use another domain in my local network and `<iframe>` on its index page to call my BB app (mainly to bypass the app portal using `returnUrl` but also to keep URL Address clean/beautiful -- to hide long app id, table rows _ids etc ) but unfortunately it does not work due to cookies behaviour in budibase, it's always unable to login/open user session. So, it might work for apps with Public access only...

**Reply by KernOil** [reply]
@shogunpurple Hi sorry missed this one. I say I am not a fan of the portal only because there is no option to make an app stand alone. currently, all users can see all published apps I have no problem with the portal if my users need access to multiple apps. however, when a user just needs one app I think it is a smoother experience if they can just enter in a subdomain something short and sweet and it will take them to the app. Some of my apps have no users it's just open to all on the network like the help desk portal or my CAD request portal. The app portal screen just adds that extra step I don't think is necessary for all applications.

**Reply by sergey-dme** [reply]
@shogunpurple Would it be possible to make a small fix for cookies:  `SameSite=None and Secure` please? It would allow users to run the app in the <iframe>. Now it works only for Public apps. Use of iframe will help with clean url and user favicon.

From update of Chromium in February 4, 2020 (Chrome 80). Cookies default to SameSite=Lax:

https://www.chromestatus.com/feature/5088147346030592

https://medium.com/trabe/cookies-and-iframes-f7cca58b3b9e

Thank you

**Comment by KernOil** [comment]
@sergey-dme yes that is what I am currently doing I copy the app's published URL to its homepage and then go to my domain's DNS settings and set up a simple subdomain forward to that long address. yes only works for public apps but will get around the portal no prob with a nice clean URL!
